packages/lib/src/utils.ts

```diff
@@ -0,0 +1,113 @@
+import { EdwardsPoint, WeierstrassPoint } from "./babyJubjub";
+import { Signature } from "./types";
+// @ts-ignore
+import { buildPoseidonReference } from "circomlibjs";
+
+export const derDecode = (encodedSig: string): Signature => {
+  const r_length = parseInt(encodedSig.slice(6, 8), 16) * 2; // Multiply by 2 to get length in hex characters
+  const s_length =
+    parseInt(encodedSig.slice(10 + r_length, 12 + r_length), 16) * 2;
+
+  const r = encodedSig.slice(8, 8 + r_length);
+  const s = encodedSig.slice(12 + r_length, 12 + r_length + s_length);
+
+  return { r: hexToBigInt(r), s: hexToBigInt(s) };
+};
+
+export const publicKeyFromString = (pubKey: string): WeierstrassPoint => {
+  if (pubKey.slice(0, 2) !== "04") {
+    throw new Error("Only handle uncompressed public keys for now");
+  }
+
+  const pubKeyLength = pubKey.length - 2;
+  const x = hexToBigInt(pubKey.slice(2, 2 + pubKeyLength / 2));
+  const y = hexToBigInt(pubKey.slice(2 + pubKeyLength / 2));
+  return new WeierstrassPoint(x, y);
+};
+
+export const hashPublicKey = async (pubKey: string): Promise<bigint> => {
+  const pubKeyPoint = publicKeyFromString(pubKey);
+  const poseidon = await buildPoseidonReference();
+  const hash = poseidon([
+    bigIntToBytes(pubKeyPoint.x),
+    bigIntToBytes(pubKeyPoint.y),
+  ]);
+
+  return hexToBigInt(poseidon.F.toString(hash, 16));
+};
+
+export const hashEdwardsPublicKey = async (
+  pubKey: EdwardsPoint
+): Promise<bigint> => {
+  const poseidon = await buildPoseidonReference();
+  const hash = poseidon([pubKey.x, pubKey.y]);
```

Comment:
Why does this not use a bigIntToBytes like the previous one?

Reply:
Removed this function